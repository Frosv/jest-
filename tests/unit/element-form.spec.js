/*
 * @Author: your name
 * @Date: 2021-01-15 09:41:12
 * @LastEditTime: 2021-01-26 09:43:21
 * @LastEditors: Please set LastEditors
 * @Description: In User Settings Edit
 * @FilePath: /vuetest/tests/unit/element-form.spec.js
 */

import {
  createLocalVue,
  shallowMount
} from '@vue/test-utils'
import elementForm from '@/views/element-form.vue'
import ElementUI from 'element-ui'
const localVue = createLocalVue()

localVue.use(ElementUI)

/**
 * 测试思路
 * 1. 活动名称
 *  a. 测试活动名称文案是否正确
 *  b. 测试活动名称输入数值是否正确获取
 *  c. 测试活动名称校验规则 超出字符，特殊字符
 * 2. 活动区域
 *  a. 模拟用户点击select
 *
 */

describe('elementForm', () => {
  const wrapper = shallowMount(elementForm, {
    localVue,
    stubs: {
      'el-form': ElementUI.Form,
      'el-form-item': ElementUI.FormItem,
      'el-input': ElementUI.Input,
      'el-select': ElementUI.Select,
      'el-option': ElementUI.Option,
      'el-button': ElementUI.Button,
      'el-date-picker': ElementUI.DatePicker
    }
  })

  it('测试快照', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('活动名称label', () => {
    expect(
      wrapper.findAll('.el-form-item__label').at(0).text()
    ).toEqual('活动名称')
  })

  // 活动名称输入
  it('活动名称输入', () => {
    wrapper.find('#actived').setValue('123')
    expect(wrapper.vm.form.name).toEqual('123')
  })

  it('模拟用户点击select', async () => {
    const el = wrapper.findComponent({
      ref: 'select'
    })
    await el.vm.$emit('fouce')
    // 显示option 找到对应li
    console.log(wrapper.find('.el-select-dropdown'))
    // const options = wrapper.find('.el-select-dropdown').findAll('li')
    // 找到第一个选项选中
    // await options.at(1).setSelected()

    // expect(wrapper.vm.form.region).toBe('shanghai')
  })

  // 只测试点击事件其实就已经完成覆盖率
  it('点击提交', () => {
    const el = wrapper.findComponent({
      ref: 'submit'
    })
    el.vm.$emit('click')
    expect(el.emitted().click).toBeTruthy()
  })
})
