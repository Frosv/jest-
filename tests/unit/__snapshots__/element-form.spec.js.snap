// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elementForm 测试快照 1`] = `
<form class="el-form">
  <div class="el-form-item">
    <label-wrap-stub><label class="el-form-item__label" style="width: 80px;">活动名称</label></label-wrap-stub>
    <div class="el-form-item__content" style="margin-left: 80px;">
      <div class="el-input">
        <!----><input type="text" autocomplete="off" id="actived" class="el-input__inner">
        <!---->
        <!---->
        <!---->
        <!---->
      </div>
      <transition-stub name="el-zoom-in-top">
        <!---->
      </transition-stub>
    </div>
  </div>
  <div class="el-form-item">
    <label-wrap-stub><label class="el-form-item__label" style="width: 80px;">活动区域</label></label-wrap-stub>
    <div class="el-form-item__content" style="margin-left: 80px;">
      <div class="el-select">
        <!---->
        <div class="el-input el-input--suffix">
          <!----><input type="text" readonly="readonly" autocomplete="off" placeholder="请选择活动区域" class="el-input__inner">
          <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-arrow-up"></i><!----><!----><!----><!----><!----></span>
          <!----></span>
          <!---->
          <!---->
        </div>
        <transition-stub name="el-zoom-in-top">
          <el-select-menu-stub transformorigin="true" placement="bottom-start" boundariespadding="0" offset="0" visiblearrow="true" arrowoffset="35" popperoptions="[object Object]" style="display: none;">
            <el-scrollbar-stub wrapclass="el-select-dropdown__wrap" viewclass="el-select-dropdown__list" tag="ul" class="" style="">
              <!---->
              <li class="el-select-dropdown__item"><span>区域一</span></li>
              <li class="el-select-dropdown__item"><span>区域二</span></li>
            </el-scrollbar-stub>
            <!---->
          </el-select-menu-stub>
        </transition-stub>
      </div>
      <transition-stub name="el-zoom-in-top">
        <!---->
      </transition-stub>
    </div>
  </div>
  <div class="el-form-item">
    <label-wrap-stub><label class="el-form-item__label" style="width: 80px;">活动时间</label></label-wrap-stub>
    <div class="el-form-item__content" style="margin-left: 80px;">
      <el-col-stub span="11" tag="div">
        <div class="el-date-editor el-input el-input--prefix el-input--suffix el-date-editor--date" style="width: 100%;">
          <!----><input type="text" autocomplete="off" name="" placeholder="选择日期" class="el-input__inner"><span class="el-input__prefix"><i class="el-input__icon el-icon-date"></i><!----></span><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-input__icon"></i><!----><!----><!----><!----></span>
          <!----></span>
          <!---->
          <!---->
        </div>
      </el-col-stub>
      <el-col-stub span="2" tag="div" class="line">-</el-col-stub>
      <el-col-stub span="11" tag="div">
        <el-time-picker-stub appendtobody="true" offset="0" boundariespadding="5" arrowoffset="35" placeholder="选择时间" clearicon="el-icon-circle-close" name="" clearable="true" id="" editable="true" align="left" value="" rangeseparator="-" validateevent="true" style="width: 100%;"></el-time-picker-stub>
      </el-col-stub>
      <transition-stub name="el-zoom-in-top">
        <!---->
      </transition-stub>
    </div>
  </div>
  <div class="el-form-item">
    <label-wrap-stub><label class="el-form-item__label" style="width: 80px;">即时配送</label></label-wrap-stub>
    <div class="el-form-item__content" style="margin-left: 80px;">
      <el-switch-stub width="40" activeiconclass="" inactiveiconclass="" activecolor="" inactivecolor="" activevalue="true" name="" validateevent="true"></el-switch-stub>
      <transition-stub name="el-zoom-in-top">
        <!---->
      </transition-stub>
    </div>
  </div>
  <div class="el-form-item">
    <label-wrap-stub><label class="el-form-item__label" style="width: 80px;">活动性质</label></label-wrap-stub>
    <div class="el-form-item__content" style="margin-left: 80px;">
      <el-checkbox-group-stub value="">
        <el-checkbox-stub label="美食/餐厅线上活动" name="type"></el-checkbox-stub>
        <el-checkbox-stub label="地推活动" name="type"></el-checkbox-stub>
        <el-checkbox-stub label="线下主题活动" name="type"></el-checkbox-stub>
        <el-checkbox-stub label="单纯品牌曝光" name="type"></el-checkbox-stub>
      </el-checkbox-group-stub>
      <transition-stub name="el-zoom-in-top">
        <!---->
      </transition-stub>
    </div>
  </div>
  <div class="el-form-item">
    <label-wrap-stub><label class="el-form-item__label" style="width: 80px;">特殊资源</label></label-wrap-stub>
    <div class="el-form-item__content" style="margin-left: 80px;">
      <el-radio-group-stub value="">
        <el-radio-stub label="线上品牌商赞助"></el-radio-stub>
        <el-radio-stub label="线下场地免费"></el-radio-stub>
      </el-radio-group-stub>
      <transition-stub name="el-zoom-in-top">
        <!---->
      </transition-stub>
    </div>
  </div>
  <div class="el-form-item">
    <label-wrap-stub><label class="el-form-item__label" style="width: 80px;">活动形式</label></label-wrap-stub>
    <div class="el-form-item__content" style="margin-left: 80px;">
      <div class="el-textarea"><textarea autocomplete="off" class="el-textarea__inner" style="min-height: -4px;"></textarea>
        <!---->
      </div>
      <transition-stub name="el-zoom-in-top">
        <!---->
      </transition-stub>
    </div>
  </div>
  <div class="el-form-item">
    <label-wrap-stub>
      <!---->
    </label-wrap-stub>
    <div class="el-form-item__content" style="margin-left: 80px;"><button type="button" class="el-button el-button--primary">
        <!---->
        <!----><span>立即创建</span>
      </button> <button type="button" class="el-button el-button--default">
        <!---->
        <!----><span>取消</span>
      </button>
      <transition-stub name="el-zoom-in-top">
        <!---->
      </transition-stub>
    </div>
  </div>
</form>
`;
